### DISCUSSION TITLE
Contributor numbers, and contribution numbers (first-time contributors compared to frequent contributors)

### DISCUSSION THREAD

**Discussion by sideshowbarker** [mdn/mdn-community#601] [Content]
From looking at the git logs today, here are a few numbers:

* 75% of our contributors are first-time contributors (574 out of 751)
* Those 574 first-time contributors account for about 15% of the total contributions (574 out of 3627)
* Our 20 most-frequent contributors account for about 65% of the total contributions (2402 out of 3627)
* The 32 people who’ve contributed 10 or more times account for 70% of the total contributions (2532 out of 3627).

Times a person has contributed | % of total people | % of contributions from people in this category
-----------------------------------------|----------:|---------------------------------------------
10 or more times                         |     < 1%  | 70%
Less than 10 times but more than 1 time  |      24% | 15%
1 time (first-time contributor)          |      75% | 15%

**Comment by Ryuno-Ki** [comment]
Would it make sense to look at these numbers over time?
(Like, monthly basis).

**Reply by sideshowbarker** [reply]
> Would it make sense to look at these numbers over time?
> (Like, monthly basis).

Yes, I reckon it would. But along with those numbers, there’s really all kind of potentially interesting stuff we could mine from the git logs. For example, we could look at data per top-level directory (api vs javascript vs html vs http, etc.).

I guess somewhere we should decide on what kind of information we’re really interested in knowing and tracking over time, and then we could add some script and GitHub Actions set up to automate generation of daily/weekly/monthly reports of some kind.

**Reply by Rumyra** [reply]
Can we make the where here? I'm very interested in tracking our github data, especially bits like this (thank you @sideshowbarker ).

I personally would like to see data like this for the org and not just content though, but I guess if we have something for content, we can port it to yari, browser-compat, interactive-examples etc...

**Reply by Rumyra** [reply]
Actually there's a discussion already about tracking contributor metrics so maybe that's a better place

https://github.com/mdn/content/discussions/5796

**Comment by sideshowbarker** [comment]
From looking at the git logs again today, here are updated numbers as of 2022-1-16:

* 77% of our contributors are first-time contributors (1203 out of 1560)
* Those 1203 first-time contributors account for about 16% of the total contributions (1203 out of 7436)
* Our 20 most-frequent contributors account for about 61% of the total contributions (4523 out of 7436	)
* The 58 people who’ve contributed 10 or more times account for 72% of the total contributions (5294 out of 7436).

Times a person has contributed | # of people | % of total people | % of contributions from people in this category
-----------------------------------------|------:|----------:|---------------------------------------------
10 or more times                                        |  58     |   4%      | 72%
Less than 10 times but more than 1 time  |   299  |   19%      | 12%
1 time (first-time contributor)                    | 1203  |     77%     | 16%

So in the last 7 months:

* The percentage of first-time contributors has increased from 75% to 77%, and the percentage of contributions they account for has increased form 15% to 16%.
* The percentage of contributors who have contributed 10 or times has increased significantly; it was less than 1% before but it’s now 4% — and the percentage of contributions from that group has increased from 70% before to 72% now.
* The percentage of changes coming from our 20 most-frequent contributors has decreased from 65% before down to 61% now — and so the percentage coming from the longer tail of contributors has increased from 35% before up to 39% now.
